index.rst

```diff
@@ -240,6 +240,101 @@ fitSrcMatchAstrometry
     :name: pl-jfb-ambitious-nofakes-01
 
 
+The Great Calibration Refactor Proposal: February 2023
+------------------------------------------------------
+
+Goals and non-goals
+"""""""""""""""""""
+
+We're replacing essentially everything between ISR and coaddition.
+ISR and coaddition will also see major changes (spurred by Calibpalooza and cell-based and chi-squared coadds, respectively) on similar timescales (we hope), but we're considering those out-of-scope and mostly orthogonal.
+While we can and should implement this piecemeal when we can, we want a complete vision of what it will look like in the end, and in some cases it may be easier to replace many tasks at once.
+
+We lean towards merging PipelineTasks with the same dimensions that run back-to-back rather than keeping them distinct.
```

Comment:
I commented on this on the main thread, but I really don't think this is the right approach. I think it would be better to spend the time on the operation side to cluster / in-memory data store. I don't really think it is appropriate at this level for task authors to design their algorithms around optimizing i/o

Reply:
I tend to agree. The clustering+in-memory data store is pretty close now (especially now that I added support for copy-on-get) but it's all the support code that is lacking such as getting BPS to configure the in-memory datastore per-task when it sees there are intermediates in the clustering, and working out which datasets have to be copied on get and which don't have to be.